radixdlt-core/radixdlt/src/main/java/com/radixdlt/api/system/health/PeersForksInfoService.java

```diff
@@ -0,0 +1,159 @@
+ * in holding or transmitting tokens by any third-party;
+ *
+ * E. breaches or failure of security including hacker attacks, loss or disclosure of
+ * password, loss of private key, unauthorised use or misuse of such passwords or keys;
+ *
+ * F. any losses including loss of anticipated savings or other benefits resulting from
+ * use of the Work or any changes to the Work (however implemented).
+ *
+ * You are solely responsible for; testing, validating and evaluation of all operation
+ * logic, functionality, security and appropriateness of using the Work for any commercial
+ * or non-commercial purpose and for any reproduction or redistribution by You of the
+ * Work. You assume all risks associated with Your use of the Work and the exercise of
+ * permissions under this License.
+ */
+
+package com.radixdlt.api.system.health;
+
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableSet;
+import com.google.inject.Inject;
+import com.radixdlt.consensus.bft.BFTNode;
+import com.radixdlt.consensus.bft.BFTValidatorSet;
+import com.radixdlt.consensus.epoch.EpochChange;
+import com.radixdlt.crypto.ECPublicKey;
+import com.radixdlt.environment.EventProcessor;
+import com.radixdlt.ledger.LedgerUpdate;
+import com.radixdlt.network.p2p.PeerEvent;
+import com.radixdlt.statecomputer.forks.Forks;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * Collects other peers' latest known forks names (received during the handshake) and compares it
+ * against local list of forks. Signals with a flag, when there's a known fork on any of the
+ * validator nodes that local node is not aware of, which means that there's potentially a newer app
+ * version to download.
+ */
+public final class PeersForksInfoService {
+  // max number of map entries (fork names)
+  private static final int MAX_FORK_KEYS = 50;
+
+  // max number of reports (peers' public keys) per fork
+  private static final int MAX_REPORTS_PER_FORK = 20;
+
+  private final Forks forks;
+
+  private BFTValidatorSet currentValidatorSet;
+  private final LinkedHashMap<String, ImmutableSet<ECPublicKey>> unknownReportedForks;
+
+  @Inject
+  public PeersForksInfoService(Forks forks, EpochChange initialEpoch) {
+    this.forks = Objects.requireNonNull(forks);
+
+    this.currentValidatorSet = initialEpoch.getBFTConfiguration().getValidatorSet();
+    this.unknownReportedForks =
+        new LinkedHashMap<>() {
+          @Override
+          protected boolean removeEldestEntry(
+              final Map.Entry<String, ImmutableSet<ECPublicKey>> eldest) {
+            return size() > MAX_FORK_KEYS;
+          }
+        };
+  }
+
+  public EventProcessor<PeerEvent> peerEventProcessor() {
+    return peerEvent -> {
+      if (peerEvent instanceof PeerEvent.PeerConnected peerConnected) {
+        final var peerChannel = peerConnected.channel();
+        peerChannel
+            .getRemoteLatestForkName()
```

Comment:
It is the former. I guess we could rename it to `remoteCandidateForkName` (and `candidateForkName` in the handshake messages), since there can only be at most a single candidate fork and, if it's there, it is always the latest one, wdyt?

Reply:
I think I like that. If the remote doesn't have a candidate fork though, this field is the most-future fixed epoch fork, right?

I guess for me, latest is just quite a confusing word, as it implies to me "most recent before now", as opposed to "furthest in the future". Perhaps `Newest` is a little clearer than `Latest`?